bit9_connector.py

```diff
@@ -462,11 +520,79 @@ def _upload_file(self, param):
             summary = action_result.update_summary({'upload_status': upload_status})
             try:
                 summary['upload_status_desc'] = self.UPLOAD_STATUS_DESCS[str(upload_status)]
-            except:
+            except Exception:
                 pass
 
         return action_result.set_status(phantom.APP_SUCCESS)
 
+    def _list_files(self, param):
+        action_result = self.add_action_result(ActionResult(param))
+        ret_val, limit = self._validate_integer(action_result, param.get('limit', 100), 'Limit', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            'limit': limit
+        }
+
+        endpoint = '/fileUpload'
+
+        ret_val, resp_json = self._make_rest_call(endpoint, action_result, method="get", params=params)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(resp_json)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully fetched available files")
+
+    def _get_file(self, param):
+        action_result = self.add_action_result(ActionResult(param))
+
+        ret_val, file_id = self._validate_integer(action_result, param["file_id"], 'File ID', False)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        params = {
+            'downloadFile': True
+        }
+
+        endpoint = '/fileUpload/' + str(file_id)
+
+        ret_val, resp_json = self._make_rest_call(endpoint, action_result, method="get")
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        filename = resp_json.get('fileName')
+
+        ret_val, resp = self._make_rest_call(endpoint, action_result, method="get", params=params)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        if hasattr(Vault, 'get_vault_tmp_dir'):
+            vault_tmp_dir = Vault.get_vault_tmp_dir()
+        else:
+            vault_tmp_dir = '/opt/phantom/vault/tmp'
+
+        file_loc = vault_tmp_dir + '/' + filename
+        with open(file_loc, 'w') as file:
+            file.write(resp.text)
+
+        vault_ret_dict = phantomrules.vault_add(container=self.get_container_id(),
```

Comment:
Can we change `vault_ret_dict` to something like `success, message, vault_id` to make it more readable?
Reference: https://docs.splunk.com/Documentation/SOARonprem/5.2.1/PlaybookAPI/VaultAPI